storage.go

```diff
@@ -138,6 +138,9 @@ func (s *Storage) read(ctx context.Context, path string, w io.Writer, opt pairSt
 	var rc io.ReadCloser
 
 	object := s.bucket.Object(rp)
+	if opt.HasSseCustomerKey {
+		object = object.Key([]byte(opt.SseCustomerKey))
```

Comment:
Maybe we should let the user pass a `[]byte` directly? Or input a string after `base64`?

Reply:
I agree, but how to support `[]byte` in the toml

Reply:
Do we need to set `CustomerKeySHA256` before we write data into it?

Reply:
> I agree, but how to support []byte in the toml

Well, it looks like not supported for now.

We can register a new type here: https://github.com/aos-dev/go-storage/blob/master/cmd/definitions/parse.go#L77

Maybe `byte_array`?

Reply:
```go
func setEncryptionHeaders(headers http.Header, key []byte, copySource bool) error {
	if key == nil {
		return nil
	}
	// TODO(jbd): Ask the API team to return a more user-friendly error
	// and avoid doing this check at the client level.
	if len(key) != 32 {
		return errors.New("storage: not a 32-byte AES-256 key")
	}
	var cs string
	if copySource {
		cs = "copy-source-"
	}
	headers.Set("x-goog-"+cs+"encryption-algorithm", "AES256")
	headers.Set("x-goog-"+cs+"encryption-key", base64.StdEncoding.EncodeToString(key))
	keyHash := sha256.Sum256(key)
	headers.Set("x-goog-"+cs+"encryption-key-sha256", base64.StdEncoding.EncodeToString(keyHash[:]))
	return nil
}
```

It looks like gcs's sdk will set it.